src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1595,7 +1595,7 @@
   </Node>
 
   <!-- The node is transformed into BoundLocal or BoundFieldAccess after inference -->
-  <Node Name="OutVariablePendingInference" Base="BoundExpression">
+  <Node Name="VariablePendingInference" Base="BoundExpression">
```

Comment:
><Node Name="VariablePendingInference" Base="BoundExpression"> [](start = 2, length = 61)

Would it make sense to have this as base class with all the implementation and still have two derived classes OutVariablePendingInference and DeconstructionVariablePendingInference? This would give us two separate node kinds and stronger verification inside compiler when each of the nodes is expected. It looks like we almost never expect both of them, so the readability of the code will be better. This will also decrease the amount of changes that should be made.